enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/LdapRealm.java

```diff
@@ -417,14 +425,22 @@ private String parseLdapServerUrl( String rawLdapServer )
 
             if ( result.hasMoreElements() )
             {
-                log.warn( String.format(
-                        "LDAP user search for user principal '%s' is ambiguous. The first match that will be checked " +
-                        "for group membership is '%s' " +
-                        "but the search also matches '%s'. Please check your LDAP realm configuration.",
-                        username,
-                        // TODO: Check if it is ok to write this potentially sensitive information to the log
-                        searchResult.toString(),
-                        ((SearchResult) result.next()).toString() ) );
+                if ( securityLog.isDebugEnabled() )
+                {
+                    securityLog.warn( withRealm(
```

Comment:
This is warning is pretty serious. I wonder if it will easily be too easily missed if we require isDebugEnabled() since that is not the default. I think it is better to leave it as a warning even though it is a configuration problem.